addon/components/ce/content-editable.ts

```diff
@@ -146,186 +83,249 @@ export default class ContentEditable extends Component<ContentEditableArgs> {
-   *
-   * Chain 1: (regular input)
-   * keydown (captured if you want to prevent the key from doing something do it here, preventdefault will prevent keypress and input from firing)
-   * keypress (we ignore this, it's badly spec'ed and behaves differently in different browsers)
-   * beforeinput (sometimes captured, has potential, because it can replace the entire keydown/keypress mess. currently only in chrome)
-   * keyup (we should ignore this, because it fires even if keydown does a preventDefault. however it is one of the few places we can capture page up, page down, arrow up and down so we capture those here using the fallback input handler)
-   * input (captured, input has happened and all you can do is clean up)
-   *
-   * Chain 2 (in FF input fires after compositionend): compositions (mostly on mac)
-   * compositionupdate (we ignore this)
-   * input (we capture this)
-   * compositionend (we capture this, if input has preventdefault this doesn't fire. not well tested)
-   *
-   * Chain 3: copy, cut, paste (may be broken if event order changes)
-   * keydown (captured, we try to ignore the event here and let it bubble)
-   * keypress (we ignore this, see above)
-   * keyup (we ignore this, see above)
-   * paste/copy/cut/undo (we capture these)
-   * input (prevent default prevents this one from triggering)
-   */
-
-  /**
-   * keyDown events are handled for simple input we take over from
-   * browser input.
-   */
   @action
   handleKeyDown(event: KeyboardEvent) {
-    if (!this.keydownMapsToOtherEvent(event)) {
-      const preventDefault = this.passEventToHandlers(event);
-      if (preventDefault) {
-        event.preventDefault();
-      }
-    }
-  }
-
-  /**
-   * keyUp events are handled for simple input we take over from
-   * browser input.
-   */
-  @action
-  handleKeyUp(event: KeyboardEvent) {
-    const preventDefault = this.passEventToHandlers(event);
-    if (preventDefault) {
-      event.preventDefault();
-    }
-  }
-
-  /**
-   * reading of blogs and part of the spec seems to indicate capture input is the safest way to capture input
-   * this method is only called for input that hasn't been handled in earlier events (like keydown)
-   */
-  @action
-  handleInput(event: InputEvent) {
-    const preventDefault = this.passEventToHandlers(event);
-    if (preventDefault) {
-      event.preventDefault();
-    }
-  }
-
-  @action
-  beforeInput(event: InputEvent) {
-    const preventDefault = this.passEventToHandlers(event);
-    if (preventDefault) {
-      event.preventDefault();
-    }
-  }
-
-  /**
-   * compositionEnd events are parsed for complex input, for uncaptured events we update
-   * the internal state to be inline with reality
-   */
-  @action
-  compositionEnd(event: CompositionEvent) {
-    const preventDefault = this.passEventToHandlers(event);
-    if (preventDefault) {
-      event.preventDefault();
-    }
-  }
-
-  @action
-  paste(event: ClipboardEvent) {
     event.preventDefault();
```

Comment:
in general I think event handling/passing needs more thought. I think we want to catch more events then we currently do here and rely more on browser events instead of just the keydown and selectionchange event.